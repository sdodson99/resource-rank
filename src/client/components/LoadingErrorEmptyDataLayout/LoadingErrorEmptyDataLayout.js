import React from 'react';
import PropTypes from 'prop-types';
import styles from './LoadingErrorEmptyDataLayout.module.css';

const LoadingErrorEmptyDataLayout = ({
  isLoading,
  loadingDisplay,
  hasError,
  errorDisplay,
  hasData,
  dataDisplay,
  noDataDisplay,
}) => (
  <div
    className={styles.LoadingErrorEmptyDataLayout}
    data-testid="LoadingErrorEmptyDataLayout"
  >
    {isLoading && loadingDisplay}

    {!isLoading && (
      <div>
        {hasError && errorDisplay}

        {!hasError && (
          <div>
            {!hasData && noDataDisplay}

            {hasData && dataDisplay}
          </div>
        )}
      </div>
    )}
  </div>
);

LoadingErrorEmptyDataLayout.propTypes = {
  isLoading: PropTypes.bool,
  hasError: PropTypes.bool,
  hasData: PropTypes.bool,
  loadingDisplay: PropTypes.node,
  errorDisplay: PropTypes.node,
  noDataDisplay: PropTypes.node,
  dataDisplay: PropTypes.node,
};

LoadingErrorEmptyDataLayout.defaultProps = {
  hasData: true,
};

export default LoadingErrorEmptyDataLayout;
