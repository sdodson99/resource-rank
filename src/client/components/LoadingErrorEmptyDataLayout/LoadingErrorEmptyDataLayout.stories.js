import React from 'react';
import LoadingErrorEmptyDataLayout from './LoadingErrorEmptyDataLayout';

export default {
  component: LoadingErrorEmptyDataLayout,
  title: 'Components/LoadingErrorEmptyDataLayout',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <LoadingErrorEmptyDataLayout {...args} />;

export const Loading = Template.bind({});
Loading.args = {
  isLoading: true,
  loadingDisplay: 'Loading...',
};

export const Error = Template.bind({});
Error.args = {
  isLoading: false,
  hasError: true,
  errorDisplay: <div className="text-red-600">Something went wrong.</div>,
};

export const Empty = Template.bind({});
Empty.args = {
  isLoading: false,
  hasError: false,
  hasData: false,
  noDataDisplay: 'No data available.',
};

export const Data = Template.bind({});
Data.args = {
  isLoading: false,
  hasError: false,
  hasData: true,
  dataDisplay: 'Successfully loaded data!',
};
