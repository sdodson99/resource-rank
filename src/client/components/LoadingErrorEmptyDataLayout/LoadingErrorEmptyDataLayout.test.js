import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import LoadingErrorEmptyDataLayout from './LoadingErrorEmptyDataLayout';
import renderer from 'react-test-renderer';

describe('<LoadingErrorEmptyDataLayout />', () => {
  it('should mount', () => {
    render(<LoadingErrorEmptyDataLayout />);

    const loadingErrorEmptyDataLayout = screen.getByTestId(
      'LoadingErrorEmptyDataLayout'
    );

    expect(loadingErrorEmptyDataLayout).toBeInTheDocument();
  });

  it('should render correctly when loading', () => {
    const tree = renderer
      .create(
        <LoadingErrorEmptyDataLayout
          isLoading={true}
          loadingDisplay={<div>Loading...</div>}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when not loading and has error', () => {
    const tree = renderer
      .create(
        <LoadingErrorEmptyDataLayout
          isLoading={false}
          hasError={true}
          errorDisplay={<div>Error</div>}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when not loading, has no error, and has no data', () => {
    const tree = renderer
      .create(
        <LoadingErrorEmptyDataLayout
          isLoading={false}
          hasError={false}
          hasData={false}
          noDataDisplay={<div>No data</div>}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when not loading, has no error, and has data', () => {
    const tree = renderer
      .create(
        <LoadingErrorEmptyDataLayout
          isLoading={false}
          hasError={false}
          hasData={true}
          dataDisplay={<div>Data</div>}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
