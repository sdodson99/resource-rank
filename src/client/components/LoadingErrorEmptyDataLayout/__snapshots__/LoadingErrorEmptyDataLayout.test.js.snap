// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingErrorEmptyDataLayout /> should render correctly when loading 1`] = `
<div
  className="LoadingErrorEmptyDataLayout"
  data-testid="LoadingErrorEmptyDataLayout"
>
  <div>
    Loading...
  </div>
</div>
`;

exports[`<LoadingErrorEmptyDataLayout /> should render correctly when not loading and has error 1`] = `
<div
  className="LoadingErrorEmptyDataLayout"
  data-testid="LoadingErrorEmptyDataLayout"
>
  <div>
    <div>
      Error
    </div>
  </div>
</div>
`;

exports[`<LoadingErrorEmptyDataLayout /> should render correctly when not loading, has no error, and has data 1`] = `
<div
  className="LoadingErrorEmptyDataLayout"
  data-testid="LoadingErrorEmptyDataLayout"
>
  <div>
    <div>
      <div>
        Data
      </div>
    </div>
  </div>
</div>
`;

exports[`<LoadingErrorEmptyDataLayout /> should render correctly when not loading, has no error, and has no data 1`] = `
<div
  className="LoadingErrorEmptyDataLayout"
  data-testid="LoadingErrorEmptyDataLayout"
>
  <div>
    <div>
      <div>
        No data
      </div>
    </div>
  </div>
</div>
`;
